### PR TITLE
replace Fuse.js with Algolia for home user search

### PR DESCRIPTION
**Pull request by Gummibeer**
**blocked by:** https://github.com/laravel/framework/pull/37080

In case you have some time @redox this is the PR to replace the current Fuse.js with Algolia & instantsearch.js
The daily re-index seems to be needed as blocked entities aren't part of the index but it's only an update call so the `laravel/scout` delete isn't called here. I will try to find a better solution.
Okay, without a custom engine this won't work as https://github.com/laravel/scout/blob/7fb1c860a2fd904f0e084a7cc3641eb1448ba278/src/Engines/AlgoliaEngine.php#L46-L73 only calls `saveObjects()`. The observer https://github.com/laravel/scout/blob/7fb1c860a2fd904f0e084a7cc3641eb1448ba278/src/ModelObserver.php#L75-L88 also makes models unsearchable if they shouldn't be searchable anymore. But not the import command.
Would like to know if it's bad practice to do a full daily re-index of the data? If it will consume my credits/units too fast? Or if that's okay? 🤔
**EDIT:** the `scout:reimport` command by `algolia/scout-extended` does the same in a lot nicer way. 🎉

apply for second category of https://github.com/algolia/algolia-coding-contest

deployment commands:
```bash
php artisan scout:sync
php artisan scout:reimport
```

### CODE REVIEW

File: app/Models/User.php [hunk 1]
```diff
@@ -228,4 +230,24 @@ public function toSitemapTag(): Url
...
             ->setChangeFrequency(Url::CHANGE_FREQUENCY_DAILY);
     }
+
+    public function toSearchableArray(): array
+    {
+        return [
+            'name' => $this->name,
+            'display_name' => $this->display_name,
+            'avatar_url' => $this->avatar_url,
+            'profile_url' => $this->profile_url,
```
Comment (redox) [inline app/Models/User.php:240]: most probably you don't want the URLs to be searchable, isn't it?

File: composer.json [hunk 2]
```diff
@@ -22,6 +23,7 @@
         "laravel/framework": "^8.12",
         "laravel/horizon": "^5.7",
         "laravel/nova": "^3.22",
+        "laravel/scout": "^8.6",
```
Comment (redox) [inline composer.json:26]: I would encourage using [scout-extended](https://www.algolia.com/doc/framework-integration/laravel/getting-started/introduction-to-scout-extended/?client=php) in order to be able to customize the specificities of Algolia.

**Review by redox — COMMENTED** [review]
Added 2 comments, but overall LGTM

File: app/Models/User.php (hunk 1 above)
Comment (Gummibeer) [reply app/Models/User.php:240]: No, but as display attribute. Do it's not configured as searchable attribute on the index.
I thought that's the way to do display only attributes. 🤔

File: composer.json (hunk 2 above)
Comment (Gummibeer) [reply composer.json:26]: Good point - will "switch". 👍

File: app/Models/User.php (hunk 1 above)
Comment (redox) [reply app/Models/User.php:240]: By default all attributes will be available for display, this is a way to configure which ones are searchable.

File: composer.json (hunk 2 above)
Comment (Gummibeer) [reply composer.json]: https://github.com/Astrotomic/opendor.me/pull/72/commits/486ba90cac351620b2c9198df6c38829b91271be

**Pull request merged**